app/controllers/saml_idp_controller.rb

```diff
@@ -58,7 +58,9 @@ def render_template_for(message, action_url, type)
   end
 
   def apply_secure_headers_override
-    use_secure_headers_override(:saml)
+    override_content_security_policy_directives(
+      form_action: ["'self'", *SecureHeadersWhitelister.whitelisted_domains]
```

Comment:
The `/test` endpoint is just lazy imo. Can we just change our tests to expect the explicit domain used in the test itself?